### PR TITLE
middleware added for validating the url

### PR DESCRIPTION
**Pull request by RoshanAnsy**
we added the new file middleware that is validates the given url first. If the url is valid then it call next functions for shorten the url.

if the url is not valid then it return to client is that entered url is not valid.
Additionally, we have also improved the code formatting

Screenshot 2024-05-16 095217

Screenshot 2024-05-16 094639

### CODE REVIEW

**Comment by abhijeetnishal** [comment]
Hey, we haven't discussed about this issue. Please don't create a PR directly it will waste you work if the issue is not necessary

**Comment by RoshanAnsy** [comment]
please schedule the meeting for it.

**Comment by abhijeetnishal** [comment]
Hey @RoshanAnsy, have you tested this is it working i.e. it is validating URL's?

**Comment by RoshanAnsy** [comment]
Yes it is working.

**Comment by abhijeetnishal** [comment]
Hey @RoshanAnsy, i just checked it's not working
Screenshot 2024-05-21 175348

**Comment by RoshanAnsy** [comment]
You are checking the request payload, but you should check the response options.

Screenshot 2024-05-22 094958

**Comment by abhijeetnishal** [comment]
But the URL I have added is invalid it should show invalid right?
Leave it request response option for now

**Comment by abhijeetnishal** [comment]
I tried with this URL https:/github.com/ but it still shortened it.

**Comment by RoshanAnsy** [comment]
when we hit the request to browsers it automatically correct certain mistakes in URLs. For example, they might interpret http:/github.com as http://github.com by adding the missing slash.

And as the result it gives you corrects url.

**Comment by abhijeetnishal** [comment]
Ok @RoshanAnsy, I will test it with other URL's and let you know. If it works then I will review the code and merge.

File: server/server.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //import express module
 const express = require("express");
-
+const dbConnect=require("./model/dbConnect")
```
Comment (abhijeetnishal) [inline server/server.js:3]: We don't need this database connection. Pull the latest code from master if conflict arises accept incoming one if you have added that code else accept current.

File: server/server.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ const corsOptions = {
   optionsSuccessStatus: 200,
 };
 
+dbConnect();
```
Comment (abhijeetnishal) [inline server/server.js:32]: Here also the same

File: server/middleware.js
Comment (abhijeetnishal) [inline server/middleware.js:1]: The middleware.js file should be present in middlewares folder with name url.js and add method name isValidURL

File: server/middleware.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+
+const https=require("node:https")
+
+async function checkWebsite(url) {
```
Comment (abhijeetnishal) [inline server/middleware.js:4]: Change method name to isValidURL

File: server/middleware.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            };
+            https.get(url,options, (res) => {
+                resolve(res);
+            }).on("error", (e) => {
+                reject(e);
+            });
+        });
+        return response.statusCode === 200;
+    } catch (error) {
+        console.error("Error checking website:", error);
```
Comment (abhijeetnishal) [inline server/middleware.js:24]: Remove this console.error, later we will use logging

File: server/middleware.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        return false;
+    }
+}
+
+
+
+exports.validateUrl=async(req,res,next)=>{
```
Comment (abhijeetnishal) [inline server/middleware.js:31]: Use as a function and later use module.exports method to export the method.

File: server/middleware.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    else{
+        return res.status(404).json("url is not valid ")
+    }
+    
+
+    }
+    catch(err){
+        console.log("err while validating the url",err);
+        return res.status(401).json("err while validating the url")
```
Comment (abhijeetnishal) [inline server/middleware.js:53]: Status code should be 500 for server related issues and json should contain object with message as you used above like this:
```
 return res.status(400).json({
                message:"provide the url",
                success:false,
```

File: server/middleware.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                success:false,
+            })
+        }
+    const check = await checkWebsite(originalUrl);
+    
+    if(check){
+        next();
+    }
+    else{
+        return res.status(404).json("url is not valid ")
```
Comment (abhijeetnishal) [inline server/middleware.js:46]: json should contain object with message as you used above like this:

```
 return res.status(400).json({
                message:"provide the url",
                success:false,
```

**Comment by RoshanAnsy** [comment]
All changes have been made as you mentioned.

**Comment by abhijeetnishal** [comment]
Hey @RoshanAnsy, resolve the conflict as I mentioned in the first comment else it will not merge.

**Comment by RoshanAnsy** [comment]
I think as you mentioned I did changes in the code base but if some conflict is there it may be due to your code base structure has changed at the major level [ I am working with your old code base may be that's the reason of conflict ].

After all that if you want to that I resolve the conflict please schedule the meeting for it I can fix it. It does not take more time I think  10 minutes is enough for it.
I think I don't have permission to resolve the conflict. Look the button of Resolve conflict is not working for me.

Screenshot 2024-05-26 132957

**Comment by abhijeetnishal** [comment]
Hey @RoshanAnsy, this happens because of other contributers add code.

If you are using vscode:
1. Pull the latest code using command: `git pull origin master`
2. In vs code it will show files with conflict then open that file and then you will get two options accept current and accept incoming. Select accept current if you have changed the code else select incoming.
3. Once done select confirm merge.
4. Finally commit the changes, it will fix this issue for sure.

**Comment by RoshanAnsy** [comment]
Have you have checked my latest commit bz for me no any conflict is visible in my vs code.

**Comment by RoshanAnsy** [comment]
I resolve the conflict please check it.

**Comment by abhijeetnishal** [comment]
Hey @RoshanAnsy, there are some issues/suggestion:
1. isValidURL is not imported in urls routes file that's why its failing to start the server.
2. Try to use prettier for code formatting to structure code.
3. Avoid adding success from response, success/failure depends on http status code.

Good luck.

Thanks for contributing.

**Pull request merged**